src/redux/actions.js

```diff
@@ -0,0 +1,252 @@
+import axios from 'axios'
+import history from './history'
+import normalize from 'jsonapi-normalizer'
+
+const reqHeaders = {
+  headers: {
+      'Content-Type': 'application/json',
+      'X-CSRF-Token': localStorage.token
+  }
+}
+
+export const signUpUserRequest = user => {
+  return dispatch => {
+    axios.post('http://localhost:3000/api/v1/signups', user,
+    { withCredentials: true })
+    .then(response => {
+      localStorage.setItem("token", response.data.csrf)
+      dispatch(loginUser(user))
+      history.push('/projects')
+    })
+    .catch(error => console.log(error))
+  }
+}
+
+export const signInUserRequest = user => {
+  return dispatch => {
+    axios.post('http://localhost:3000/api/v1/signins', user,
+    { withCredentials: true })
+    .then(response => {
+      localStorage.setItem("token", response.data.csrf)
+      dispatch(loginUser(user))
+      history.push('/projects')
+    })
+    .catch(error => console.log(error))
+  }
+}
+
+export const logoutUserRequest = () => {
+  return dispatch => {
+    localStorage.removeItem('token')
+    dispatch(logoutUser())
+    history.push('/signin')
+  }
+}
+
+export const fetchProjectsRequest = () => {
+  return dispatch => {
+    axios.get('http://localhost:3000/api/v1/projects',
+    { withCredentials: true })
+    .then(response => {
+      dispatch(projectsRequestSuccess(normalize(response.data)))
+    })
+    .catch(error => console.log(error))
+  }
+}
+
+export const createProjectRequest = (project) => {
+  return dispatch => {
+    axios('http://localhost:3000/api/v1/projects', {
+      method: "post",
+      data: project,
+      withCredentials: true,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': localStorage.token
+      }
+    })
+    .then(response => {
+      dispatch(projectCreateRequestSuccess(normalize(response.data)))
+    })
+    .catch(error => console.log(error))
+  }
+}
+
+export const removeProjectRequest = (project_id) => {
+  return dispatch => {
+    axios('http://localhost:3000/api/v1/projects/' + project_id, {
+      method: "delete",
+      withCredentials: true,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': localStorage.token
+      }
+    })
+    .then(response => {
+      dispatch(projectDeleteRequestSuccess(project_id))
+    })
+    .catch(error => console.log(error))
+  }
+}
+
+export const updateProjectRequest = (project, project_id) => {
+  return dispatch => {
+    axios('http://localhost:3000/api/v1/projects/' + project_id, {
```

Comment:
please try to create axios [instance](https://github.com/axios/axios#custom-instance-defaults), and set headers and options there, to dry your code